Assignments/Splitwise/Python/Group.py

```diff
@@ -0,0 +1,188 @@
+from typing import List
+from User import User
+from Expense import Expense
+
+class Group: 
+    def __init__(self, inputList: List[str]): 
+        """
+        Constructor : defines the Group class and adds all the users in the group.
+
+        Parameters:
+            inputList (list(str)) : list of all words from command line
+
+        Returns : 
+            None
+        """
+        self.users = {}
+        self.balances = {}
+        self.expenses = []
+
+        usersString = inputList[2:]
+        count = 1
+        for usr in usersString:
+            id = 'u'+str(count)
+            self.users[id] = User(id,usr)
+            count += 1
+
+    def addExpense(self, inputList: List[str]):
+        """
+        Adds the expense from command line. Updates each user's balances.
+        Adds the expense to each participating user's passbook.
+
+        Parameters:
+            inputList (list(str)) : list of all words from command line
+        
+        Returns:
+            None
+
+        """
+
+        expenseBy = inputList[1]
+        amountToSplit = int(inputList[2])
+        splitAmong = []
+        splitArgs = []
+        expenseType = ""
+
+        amount = amountToSplit
+        num = int(inputList[3])
+        owedBy = inputList[4:4+num]
+
+        for user in owedBy:
+            splitAmong.append(self.users[user])
+
+        splitAmong.append(self.users[expenseBy])
+        splitAmong = set(splitAmong)
+
+        expenseType = inputList[4+num]
+
+        # Update the transaction 
+        def updateAmount(owedTo, usr, amt) :
```

Comment:
Why is this a nested function. Don't you want others to use it ?. What is the intention behind this ?

Reply:
This function was only called when there's an expense added. So there's no other function calling it. I thought it might be fitting to make it nested.